uaclient/util.py

```diff
@@ -687,3 +690,20 @@ def redact_sensitive_logs(
 def should_reboot() -> bool:
     """Check if the system needs to be rebooted."""
     return os.path.exists(REBOOT_FILE_CHECK_PATH)
+
+
+def wrap_text(
+    text: str, width: int = DEFAULT_PRINT_WRAP_WIDTH, **kwargs
+) -> str:
+    """
+    Wrap text to 80 characters by default. Result is ready to be printed.
+
+    Additional kwargs are passed on to textwrap.wrap
+
+    @param text: The long string that needs to be printed but wrapped at a
+                    given width.
+    @param width: The maximum line length allowed in the result.
+    @return: The text with newlines inserted appropriately so that it never
+                    exceeds the given width.
+    """
+    return "\n".join(textwrap.wrap(text, width=width, **kwargs))
```

Comment:
While docs.python.org doesn't include docs for `textwrap` for python 3.4, I did check that this is available in a trusty container on python3.4.

This helper function only really adds two things for convenience. It sets the width to 80 (70 is the default), and joins the result with newlines so we get a string ready to be printed rather than a list of strings.

Reply:
Normally, I'd want to avoid declaring our own func for something that currently wraps a single line of code shipped by a standard library, but in uaclient case, I think we will have extra concerns to worry about such as how we want to indent across wraps and the potential of colorized character code sequences. It's because of the latter that I think we may need to have this function in util.   That said, I think we might need to look at baking some of this utility into status.colorize_commands too.

Reply:
Yeah I generally agree - the only reason I opted for this function is to replace the default width so that we don't have to pass that every time we use it, but that's a really minor thing. I'm totally open to removing this entirely and we can just call `textwrap.wrap(whatever, width=util.DEFAULT_WRAP_WIDTH)` wherever we need it.

I was thinking more about the colorization, and I'm leaning towards suggesting that we always wrap before we colorize, to avoid problems like you were mentioning (as opposed to getting the wrap function to understand the color control chars). Do you think that is workable or do you know of scenarios where we definitely have to colorize a chunk of text _before_ we wrap it?

I need to look more into how `status.colorize_commands` is used, but yes it looks like we'll probably want a textwrap on that `content` var before adding the color control codes around it.

Reply:
> Yeah I generally agree - the only reason I opted for this function is to replace the default width so that we don't have to pass that every time we use it, but that's a really minor thing. I'm totally open to removing this entirely and we can just call `textwrap.wrap(whatever, width=util.DEFAULT_WRAP_WIDTH)` wherever we need it.

Yeah that makes sense too minimally, it'd be nice to avoid reflecting the same additional params to every callsite.

> 
> I was thinking more about the colorization, and I'm leaning towards suggesting that we always wrap before we colorize, to avoid problems like you were mentioning (as opposed to getting the wrap function to understand the color control chars). Do you think that is workable or do you know of scenarios where we definitely have to colorize a chunk of text _before_ we wrap it?

Excellent thought/suggestion. yes we should be able to wrap before colorization in colorize_commands callsites. or rather inside the colorize_commands call itself.  I think generally this stance should hold +1 on the suggestion, so maybe this isn't as big as I thought.


> 
> I need to look more into how `status.colorize_commands` is used, but yes it looks like we'll probably want a textwrap on that `content` var before adding the color control codes around it.

+1 go for it.